### PR TITLE
elixir: add atom exhausting rule

### PR DESCRIPTION
**Pull request by wingyplus**
Detect all Elixir `(String|List).to_atom/1` function calls and fix it with `(String|List).to_existing_atom/1`.

### CODE REVIEW

**Comment by wingyplus** [comment]
Move to correctness category from suggestion in Slack thread https://semgrep.slack.com/archives/C01LNF69F9B/p1692552800141539

File: elixir/lang/correctness/atom-exhaustion.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+rules:
+  - id: atom_exhaustion
+    message: >-
+      Atom doesn't garbage by the Garbage Collector. Creating it dynamically with `String.to_atom/1` or `List.to_atom/1` can possible memory leaks if the input cannot control.
```
Comment (colleend) [inline elixir/lang/correctness/atom-exhaustion.yaml]:
```suggestion
      Atom values are appended to a global table but never removed. If input is user-controlled, dynamic instantiations such as `String.to_atom` or `List.to_atom` can lead to possible memory leaks. Instead, use `String.to_existing_atom` or `List.to_existing_atom`.
```

**Review by colleend — CHANGES_REQUESTED** [review]
Thank you for submitting this! Changed the message a bit, but otherwise it looks good to me 👍

**Comment by wingyplus** [comment]
> Thank you for submitting this! Changed the message a bit, but otherwise it looks good to me 👍

Thank you. The message looks better!

**Comment by wingyplus** [comment]
@colleend done. :)

**Review by colleend — APPROVED** [review]
lgtm!

**Pull request merged**